osu.Game/Rulesets/RulesetStore.cs

```diff
@@ -129,13 +128,18 @@ private void addMissingRulesets()
                 {
                     try
                     {
-                        var instanceInfo = ((Ruleset)Activator.CreateInstance(Type.GetType(r.InstantiationInfo).AsNonNull())).RulesetInfo;
+                        var resolvedType = Type.GetType(r.InstantiationInfo);
 
-                        r.Name = instanceInfo.Name;
-                        r.ShortName = instanceInfo.ShortName;
-                        r.InstantiationInfo = instanceInfo.InstantiationInfo;
+                        if (resolvedType != null)
```

Comment:
Shouldn't a `null` here cause `r.Available` to be set to `false`? I think it won't be with these changes, will it?

Reply:
You're welcome to confirm, but this should be in the `addMissingRuleset` path where I believe this is already guaranteed.

Maybe an assert would be nice to have?

Reply:
I manually changed my DB to the following:

![2021-11-11-074618_1208x201_scrot](https://user-images.githubusercontent.com/20418176/141251076-669201eb-8433-4ff8-b914-5e97f4dbb254.png)

and ran main game where `EmptyFreeformRuleset` was not available. The `Available` flag is databased, was already 1, and did not update to 0, which caused a crash at `RealmKeyBindingStore`.

Reply:
Guess we need some better test coverage here, scary. Method names should not be trusted.